src/hotspot/share/opto/vtransform.hpp

```diff
@@ -478,28 +483,40 @@ class VTransformReductionVectorNode : public VTransformVectorNode {
   NOT_PRODUCT(virtual const char* name() const override { return "ReductionVector"; };)
 };
 
-class VTransformLoadVectorNode : public VTransformVectorNode {
+class VTransformMemVectorNode : public VTransformVectorNode {
+private:
+  const VPointer _vpointer; // with size of the vector
```

Comment:
Note: make sure we can store the size-adjusted `VPointer` for load/store vectors.